include/bluetooth/gatt.h

```diff
@@ -642,8 +642,8 @@ ssize_t bt_gatt_attr_read_cpf(struct bt_conn *conn,
  *  @param _value Descriptor attribute value.
  */
 #define BT_GATT_CPF(_value)						\
-	BT_GATT_ATTRIBUTE(BT_UUID_GATT_CPF, BT_GATT_PERM_READ,		\
-			  bt_gatt_attr_read_cpf, NULL, _value)
+	BT_GATT_DESCRIPTOR(BT_UUID_GATT_CPF, BT_GATT_PERM_READ,		\
+			  bt_gatt_attr_read_cpf, NULL, (void * const)_value)
```

Comment:
why is casting needed here?

Reply:
Ah, I forgot that I had that change baked in here as well. The Characteristic Presentation Format is in practice always constant, and the read callback will never modify it, so this cast allows the struct to be declared const and thereby put into ROM instead of RAM.

Reply:
If you insist, I can split this up to a separate commit, or clarify the issue in the existing commit message.

Reply:
So what I have is something like this:
```
static const struct bt_gatt_cpf name_cpf = {
        .format = CPF_FORMAT_UTF8,
};

static struct bt_gatt_attr name_attrs[] = {
        ...
        BT_GATT_CPF(&name_cpf),
```

And without the cast this results in:
```
warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
```